lib/node_modules/@stdlib/math/generics/random/sample/lib/factory.js

```diff
@@ -239,20 +268,20 @@ function factory() {
 		} else {
 			size = pool.length;
 		}
-
 		if (
 			replace === false &&
 			size > pool.length
 		) {
 			throw new RangeError( 'invalid input option. `size` option must be less than or equal to the population size when `replace` is `false`. Option: `' + size + '`.' );
 		}
-
-		if ( replace !== false ) {
+		if ( replace ) {
 			return discreteUniform( pool, size, rand );
 		}
 		ret = fisherYates( pool, size, rand );
-		// Slice off observations included in `ret`...
-		pool = pool.slice( size, pool.length );
+		if ( mutate ) {
+			// Slice off observations included in `ret`...
```

Comment:
Is it? Without it, it is not apparent that the elements removed from `pool` are those that will get returned in `ret`. You need to look through `fisherYates` to deduce this.